app/knight.py

```diff
@@ -0,0 +1,126 @@
+from __future__ import annotations
+
+
+class Knight:
+
+    def __init__(self, knight_dict: dict) -> None:
+        self.name = knight_dict["name"]
+
+        self.armour = knight_dict["armour"]
+        self.weapon = knight_dict["weapon"]
+        self.potion = knight_dict["potion"]
+
+        self.hp = knight_dict["hp"]
+        self.power = knight_dict["power"] + self.weapon["power"]
+        self.protection = 0
+
+        for arm in self.armour:
+            self.protection += arm["protection"]
+
+        if self.potion is not None:
+            if "power" in self.potion["effect"]:
+                self.power += self.potion["effect"]["power"]
+
+            if "protection" in self.potion["effect"]:
+                self.protection += self.potion["effect"]["protection"]
+
+            if "hp" in self.potion["effect"]:
+                self.hp += self.potion["effect"]["hp"]
+
```

Comment:
The same, read the checklist

![image](https://user-images.githubusercontent.com/104218550/219118619-778be7e4-9e2c-4e7b-974c-63f797a77b86.png)

I give you a hint: use `setattr `